prov/psm/src/psmx_init.c

```diff
@@ -398,7 +399,7 @@ PSM_INI
 	int check_version;
 	int err;
 
-	fi_log_init();
+	psm_handle = fi_log_register_provider("psm");
```

Comment:
@shefty 
I don't know if you care about providers being able to use the logging macros inside of their INI functions, but if you don't I think we can get away without needing extra calls in the provider. There is still the issue of how to get the log handle to the providers.